test/Microsoft.Diagnostics.EventFlow.Outputs.Tests/ElasticSearchOutputTests.cs

```diff
@@ -86,9 +88,40 @@ public void VerifyAllConfigOptionsMappedToConfigObject()
                 Assert.Equal("verysecret", esOutputConfiguration.Proxy.UserPassword);
 
                 Assert.Equal("date_nanos", esOutputConfiguration.Mappings.Properties["timestamp"].Type);
+
+                var eso = new ElasticSearchOutput(esOutputConfiguration, new CustomHealthReporter(configuration));
+                Assert.Equal($"myprefix-{DateTimeOffset.UtcNow:yyyy.MM}", eso.GetIndexName(eso.connectionData));
             }
         }
 
+        [Theory]
+        [InlineData(2023, 1, 1, "yyyy.q", "2023.q1")]
+        [InlineData(2023, 3, 1, "yyyy.q", "2023.q1")]
+        [InlineData(2023, 4, 1, "yyyy.q", "2023.q2")]
+        [InlineData(2023, 1, 1, "yyyy.MM.w", "2023.01.w1")]
+        [InlineData(2023, 2, 1, "yyyy.MM.w", "2023.02.w1")]
+        [InlineData(2023, 1, 11, "yyyy.MM.w", "2023.01.w2")]
+        [InlineData(2023, 2, 12, "yyyy.MM.w", "2023.02.w3")]
+        [InlineData(2023, 2, 1, "yyyy.W", "2023.w5")]
+        [InlineData(2023, 2, 1, "yyyy.WW", "2023.w05")]
+        [InlineData(2023, 4, 1, "yyyy.W", "2023.w13")]
+        [InlineData(2023, 4, 1, "yyyy.WW", "2023.w13")]
+        [InlineData(2023, 4, 2, "yyyy.WW", "2023.w14")]
```

Comment:
Consider adding a few more tests for more-than-one special replacement chars in a pattern. For example, with `yyyy.q.WW` as a pattern I would expect something like "2023.q1.w05" as the result of formatting 2/1/2023.

Reply:
Sorry, the proverbial fecal matter was hitting the rotating blades at work yesterday.
Sure, I'll get some more tests added.

Reply:
Finding the indexes early on causes issues because the position changes as I update the string. I either need to find positions live, or process the indexes in reverse order, highest found index first. I'll try that.

Reply:
I had to jump through some hoops to make sure this was still performant, but it is working now and checked in.

Honestly, if you would rather I rip it all out and just specify that only one special formatting character is allowed, I wouldn't push back, but it IS working. :-)

Reply:
No, no, I love it!